plan/planbuilder.go

```diff
@@ -407,13 +408,84 @@ func (b *planBuilder) buildPrepare(x *ast.PrepareStmt) Plan {
 	return p
 }
 
+func (b *planBuilder) buildCheckIndex(dbName model.CIStr, as *ast.AdminStmt) Plan {
+	tblName := as.Tables[0]
+	tbl, err := b.is.TableByName(dbName, tblName.Name)
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+	tblInfo := tbl.Meta()
+
+	// get index information
+	var idx *model.IndexInfo
+	for _, index := range tblInfo.Indices {
+		if index.Name.L == as.Index {
+			idx = index
+			break
+		}
+	}
+	if idx == nil {
```

Comment:
Do we need to check if the index state is public?

Reply:
Done.